### PR TITLE
[Feature-400] Tequilapi package upgrade

### PR DESCRIPTION
**Pull request by ignasbernotas**
Currently blocked by: https://github.com/mysteriumnetwork/js-tequilapi/pull/31

### CODE REVIEW

File: test/unit/app/data-fetchers/tequilapi-proposal-fetcher.spec.js [hunk 1]
```diff
@@ -54,14 +55,14 @@ describe('TequilapiProposalFetcher', () => {
...
+    parseProposalDTO({
+      id: 1,
       providerId: '0x1',
       serviceType: 'openvpn',
       serviceDefinition: {}
     }),
-    new ProposalDTO({
-      id: '0x2',
+    parseProposalDTO({
+      id: 2,
```
Comment (donce) [inline test/unit/app/data-fetchers/tequilapi-proposal-fetcher.spec.js]: TIL: these ids are not globally unique - unique proposal id is `id`+`providerId`+`serviceType`.
Would suggest to leave this id non-unique intentionally to make this more clear.

```suggestion
      id: 1,
```

File: test/unit/app/data-fetchers/tequilapi-proposal-fetcher.spec.js [hunk 2]
```diff
@@ -113,16 +114,18 @@ describe('TequilapiProposalFetcher', () => {
...
+        id: 1,
         providerId: '0x1',
         serviceType: 'openvpn',
-        serviceDefinition: {}
+        serviceDefinition: {},
+        metrics: undefined
       })
       expect(proposals[1]).to.deep.equal({
-        id: '0x2',
+        id: 2,
```
Comment (donce) [inline test/unit/app/data-fetchers/tequilapi-proposal-fetcher.spec.js]: TIL: these ids are not globally unique - unique proposal id is `id`+`providerId`+`serviceType`.
Would suggest to leave this id non-unique intentionally to make this more clear.

```suggestion
      id: 1,
```

File: src/renderer/store/modules/identity.js [hunk 3]
```diff
@@ -42,7 +42,7 @@ function mutationsFactory (bugReporter: BugReporter, communication: RendererComm
     [type.SET_CURRENT_IDENTITY] (state, identity: IdentityDTO) {
       state.current = identity
       bugReporter.setUser(identity)
-      communication.currentIdentityChanged.send(identity)
+      communication.currentIdentityChanged.send({ id: identity.id || '' })
```
[the remark this answers is not in the document]
Comment (ignasbernotas) [reply src/renderer/store/modules/identity.js]: This is now handled here: https://github.com/mysteriumnetwork/mysterium-vpn/pull/408/commits/eb9efcccdbe94bed4399b9711df9ff79bf13c5d3#diff-fbd6c1aba00be1f1b18c24f7f0d65bb6R51 but this is needed to workaround until identityDto is fixed in lib.

File: src/renderer/store/modules/identity.js [hunk 4]
```diff
@@ -42,6 +42,7 @@ function mutationsFactory (bugReporter: BugReporter, communication: RendererComm
     [type.SET_CURRENT_IDENTITY] (state, identity: IdentityDTO) {
       state.current = identity
       bugReporter.setUser(identity)
+      // @TODO remove check
```
Comment (donce) [inline src/renderer/store/modules/identity.js]: Where is this @ convention coming from? :O Most of our TODOs are ust `// TODO`

File: src/app/identity-manager.js [hunk 5]
```diff
@@ -49,6 +49,12 @@ class IdentityManager {
   }
 
   setCurrentIdentity (identity: IdentityDTO) {
+    if (!identity.id) {
+      const message = 'Cannot set empty identity.'
+
+      throw new Error(message)
```
Comment (donce) [inline src/app/identity-manager.js]: variable looks like overkill:
```js
throw new Error('Cannot set empty identity.')
```

**Pull request merged**